packages/mantine/src/components/collection/Collection.tsx

```diff
@@ -102,6 +98,25 @@ interface CollectionProps<T>
      * @default false
      */
     required?: boolean;
+    /**
+     * Function that changes the list of values order.
+     *
+     * @param payload The origin and destination index of the item to reorder
+     */
+    reorderListItem?: (payload: Record<'from' | 'to', number>) => void;
```

Comment:
Can we import the payload type directly from mantine?

```suggestion
    reorderListItem?: (payload: ReorderPayload) => void;
```